### PR TITLE
Add support for eCash staking rewards

### PR DESCRIPTION
**Pull request by Fabcien**
Use the output from the getblocktemplate RPC to build the staking reward output.

### CODE REVIEW

File: src/stratifier.c [hunk 1]
```diff
@@ -625,7 +628,7 @@ static void generate_coinbase(ckpool_t *ckp, workbase_t *wb)
 	/* Coinb2 address goes here, takes up 23~25 bytes + 1 byte for length */
 
 	wb->coinb3len = 0;
-	wb->coinb3bin = ckzalloc(256 + wb->insert_witness * (8 + witnessdata_size + 2));
+	wb->coinb3bin = ckzalloc(512 + wb->insert_witness * (8 + witnessdata_size + 2));
```
Comment (BytesOfMan) [inline src/stratifier.c:631]: 512 instead of 256 since adding another output?

Comment (Fabcien) [reply src/stratifier.c:631]: yes it gives some room at the expense of 256 bytes of memory

**Comment by BytesOfMan** [comment]
Are these changes tested in this repo / what would a test plan look like here?

**Comment by Fabcien** [comment]
This is legacy code with no unit test, but it's been tested by running it on testnet.

**Review by BytesOfMan — APPROVED** [review]

**Pull request merged**